lisa/transformers/upgrade_packages.py

```diff
@@ -119,7 +119,9 @@ def validate(self) -> None:
         )
 
     def install(self) -> None:
-        self._update_repo()
+        if self.runbook.repo_url.strip():
+            self._update_repo()
```

Comment:
how is the repo_url used?

Reply:
It specifies which repo to take the updates from. It might also be used in the future to specify a ppa repo. If no repo url is specified, then it will not change the repo list.

Reply:
The logic is confusing, if the repo_url is not used. Can it be changed to a meaningful flag?

Reply:
I feel like this logic makes sense: only update the repo list if a repo url is provided.

Reply:
If the repo url is used as a string, it makes sense. But it's not used, so it's to use a string empty or not to run logic or not. This should be done by a bool value.

Reply:
The repo url is used as a string in the `_update_repo` function

Reply:
I see. I missed it...